src/Features/CSharp/Portable/Completion/CompletionProviders/ObjectCreationCompletionProvider.cs

```diff
@@ -79,6 +79,15 @@ protected override (string displayText, string insertionText) GetDisplayAndInser
                 return (symbol.Name, symbol.Name);
             }
 
+            // If not all  of the type arguments are concrete types, eg
+            // we've inferred something like `Task<T>`, just suggest the
+            // generic type `Task<>`
+            if (symbol is INamedTypeSymbol nt &&
+                nt.TypeArguments.Any(t => t.TypeKind == TypeKind.TypeParameter))
+            {
+                return CompletionUtilities.GetDisplayAndInsertionText(symbol, context);
```

Comment:
Maybe not intuitively, but probably some reading the code. The base method does some non-language specific stuff while the CompletionHelpers classes are language-specific. Perhaps something you would only know if you were familiar with completion.